### PR TITLE
Begin Work for state tracking

### PR DESCRIPTION
**Pull request by sithlord48**
Initial work to track the state of the printer. A few imediate beinfits have already been reaslized you can connect and disconnect now successfully multipal times durring runtime. And simple Pause feature is working.

### CODE REVIEW

File: testclient/mainwindow.cpp [hunk 1]
```diff
@@ -199,12 +200,37 @@ void MainWindow::fanSpeedPBClicked()
...
+    case DISCONNECTED:
+        QMessageBox::information(this, tr("Error"), tr("Not Connected To a Printer"));
+        break;
+
+    case CONNECTING:
+        QMessageBox::information(this, tr("Error"), tr("Firmware Plugin not loaded try sending M115"));
+        break;
+
+    case IDLE:
+        fileName = QFileDialog::getOpenFileName(this, tr("Select A file to print"), QDir::homePath(), "*.gcode");
```
Comment (patrickelectric) [inline testclient/mainwindow.cpp]: a file

File: testclient/mainwindow.cpp [hunk 2]
```diff
@@ -199,12 +200,37 @@ void MainWindow::fanSpeedPBClicked()
...
+        } else {
+            addLog(tr("Print: %1").arg(fileName));
+            ui->printPB->setText(tr("Pause Print"));
+            core->print(fileName);
+        }
+        break;
+
+    case BUSY:
+        core->setState(PAUSE);
+        ui->printPB->setText(tr("Resume Print"));
```
Comment (patrickelectric) [inline testclient/mainwindow.cpp:227]: Printer

File: testclient/mainwindow.cpp [hunk 3]
```diff
@@ -199,12 +200,37 @@ void MainWindow::fanSpeedPBClicked()
...
+        break;
+
+    case PAUSE:
+        core->setState(BUSY);
+        ui->printPB->setText(tr("Pause Print"));
```
Comment (patrickelectric) [inline testclient/mainwindow.cpp:232]: Printer

File: testclient/mainwindow.ui [hunk 4]
```diff
@@ -6,8 +6,8 @@
    <rect>
```
Comment (patrickelectric) [inline testclient/mainwindow.ui:6]: ui changes  in different commit please

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: testclient/mainwindow.cpp (hunk 2 above)
Comment (sithlord48) [reply testclient/mainwindow.cpp:227]: Its Print as in the current print job running.

File: testclient/mainwindow.ui (hunk 4 above)
Comment (sithlord48) [reply testclient/mainwindow.ui:6]: Just the changes to the ui file in a seperate commit?

File: testclient/mainwindow.cpp (hunk 1 above)
Comment (sithlord48) [reply testclient/mainwindow.cpp]: :heavy_check_mark:  done

File: testclient/mainwindow.ui (hunk 4 above)
Comment (sithlord48) [reply testclient/mainwindow.ui:6]: :heavy_check_mark:  done

**Review by patrickelectric — APPROVED** [review]
Can be merged

**Pull request merged**